module/Photo/src/Photo/Controller/PhotoController.php

```diff
@@ -14,8 +14,35 @@ public function indexAction()
         $albums = $album_service->getAlbums();
         $photo_service = $this->getPhotoService();
 
+        //we'll fix this ugly thing later vv
+        $config = $this->getPhotoService()->getConfig();
+        $basedir = str_replace("public", "", $config['upload_dir']);
+        $albums = $photo_service->populateCoverPhotos($albums);
         return new ViewModel(array(
-            'albums' => $albums
+            'albums' => $albums,
+            'basedir' => $basedir
+        ));
+    }
+
+    /**
+     * Called on viewing a photo
+     * 
+     */
+    public function viewAction()
+    {
+        $photo_id = $this->params()->fromRoute('photo_id');
+        $photo_service = $this->getPhotoService();
+        $photo = $photo_service->getPhoto($photo_id);
+        $config = $photo_service->getConfig();
+        $next = $photo_service->getNextPhoto($photo);
+        $previous = $photo_service->getPreviousPhoto($photo);
```

Comment:
Het is waarschijnlijk beter om hier een service method te maken die al deze dingen doet, en dan een array hiervoor teruggeeft. (deze kan natuurlijk intern deze methoden aanroepen)